### DISCUSSION TITLE
Utilisation de MycilaJSY

### DISCUSSION THREAD

**Discussion by francois57** [mathieucarbou/MycilaJSY#7] [General]
Bonjour.
En compilant un exemple (n'importe lequel) de MycilaJSY, j'ai cette erreur,

E:\GitHub\JSY\JSY.ino:6:9: error: 'JSY' in namespace 'Mycila' does not name a type
 Mycila::JSY jsy;
         ^~~
In file included from E:\GitHub\JSY\JSY.ino:4:
e:\GitHub\libraries\MycilaJSY\src/MycilaJSY.h:110:19: note: 'Mycila::JSY' declared here
   extern JSYClass JSY;
                   ^~~
E:\GitHub\JSY\JSY.ino: In function 'void setup()':
E:\GitHub\JSY\JSY.ino:14:3: error: 'jsy' was not declared in this scope
   jsy.begin(Serial2, 16, 17);
   ^~~
E:\GitHub\JSY\JSY.ino: In function 'void loop()':
E:\GitHub\JSY\JSY.ino:18:7: error: 'jsy' was not declared in this scope
   if (jsy.read()) {
       ^~~
E:\GitHub\JSY\JSY.ino:25:3: error: 'jsy' was not declared in this scope
   jsy.resetEnergy();
   ^~~

Utilisation de la bibliothèque ArduinoJson version 7.1.0 dans le dossier: E:\GitHub\libraries\ArduinoJson 
Utilisation de la bibliothèque MycilaJSY version 3.0.1 dans le dossier: E:\GitHub\libraries\MycilaJSY 
Utilisation de la bibliothèque MycilaLogger version 3.1.1 dans le dossier: E:\GitHub\libraries\MycilaLogger 
exit status 1

Compilation error: 'JSY' in namespace 'Mycila' does not name a type

avez vous une solution 
Merci

**Comment by mathieucarbou** [comment]
Bonjour,
Les exemples compilent puisqu'ils sont testés automatiquement à chaque modification du projet par le build continu, à la fois avec Arduino CI et PlatformIO.
Il y a donc sûrement une erreur de setup sans votre environnement.
Je n'utilise pas Arduino IDE car ce n'est pas in vrai IDE de développement, j'utilise PlatformIO et Vscode. 
Donc je ne saurais pas vous conseiller quoi faire pour résoudre votre problème d'environnement.
Mais vu l'erreur, il semblerait que la librairie n'est pas installée (qu'elle ne fait pas partie de la compilation) puisque les classes et namespace ne sont pas trouvés.
Cdlt,
Mathieu.

**Comment by francois57** [comment]
Bonjour,
Ok je vais creuser dans ce sens
Merci
François
